src/urllib3/util/retry.py

```diff
@@ -270,7 +277,15 @@ def get_retry_after(self, response):
         if retry_after is None:
             return None
 
-        return self.parse_retry_after(retry_after)
+        retry_after = self.parse_retry_after(retry_after)
+
+        if self.max_retry_wait_length != -1 and retry_after > self.max_retry_wait_length:
+            raise ExceedingWaitTime(
```

Comment:
Instead of raising an exception what are your thoughts on limiting the amount of time that would be slept in the `sleep_for_retry` method? I guess I don't understand the use-case for raising here.